pkg/backend/snapshot.go

```diff
@@ -249,18 +242,30 @@ func (ssm *sameSnapshotMutation) End(step deploy.Step, successful bool) error {
 	contract.Assert(successful)
 	logging.V(9).Infof("SnapshotManager: sameSnapshotMutation.End(..., %v)", successful)
 	return ssm.manager.mutate(func() bool {
+		sameStep := step.(*deploy.SameStep)
+
 		ssm.manager.markDone(step.Old())
+
+		// In the case of a 'resource create' in a program that wasn't specified by the user in the
+		// --target list, we *never* want to write this to the checkpoint.  We treat it as if it
+		// doesn't exist at all.  That way when the program runs the next time, we'll actually
+		// create it.
+		if sameStep.IsSkippedCreate() {
+			return false
+		}
+
 		ssm.manager.markNew(step.New())
```

Comment:
the `markNew` was the line that was ultimately causing problems.  Even though we didn't think we needed to write this information to the snapshot (below), we still added the 'create' to the 'news' list which ended up still writing later on